pkg/volume/secret/secret_test.go

```diff
@@ -40,12 +43,15 @@ import (
 )
 
 func TestMakePayload(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.UserNamespacesSupport, true)()
```

Comment:
I think I'd like to see the cases for UserNamespacesSupport broken into a seperate test (`TestMakePayload_UserNamespaces`) with cases for the feature being enabled and also disabled (partially to ensure gating, but also to help prevent regression)

Reply:
Fixed too!